etc/picongpu/taurus-tud/knl_picongpu.profile.example

```diff
@@ -0,0 +1,42 @@
+# Text Editor for Tools #######################################################
+#   - examples: "nano", "vim", "emacs -nw", "vi" or without terminal: "gedit"
+#export EDITOR="nano"
+
+# General modules #############################################################
+#
+module purge
+module load cmake/3.9.0
+module load git
+module load intelmpi/2017.2.174
+
+# Compilers ###################################################################
+### ICC
+module load intel/2017.2.174
+# Boost needs to be build yourself for now!
+# The boost root path needs to be adjusted.
+export BOOST_ROOT=<BOOST_INSTALLATION_DIR>
+export LD_LIBRARY_PATH=$BOOST_ROOT:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=$BOOST_ROOT/lib:$LD_LIBRARY_PATH
+
+export PICSRC=$HOME/src/picongpu
+export PIC_BACKEND="omp2b:native"
+export PIC_PROFILE=$(cd $(dirname $BASH_SOURCE) && pwd)"/"$(basename $BASH_SOURCE)
```

Comment:
This line is not working on the knl node:
```
dirname: missing operand
Try 'dirname --help' for more information.
basename: missing operand
Try 'basename --help' for more information
```
I don't know why but `$BASH_SOURCE` is empty

Reply:
waaaaat? :'(

Reply:
This is only an issue on the KNL node. The variable is set on the login node.

Reply:
not too crucial, since it is only needed on the head node as long as the compute node is not re-submitting.

does it also throw into stderr when running a job non-interactively?

this might be, since interactive commands need to be started with `srun ...` after allocation (e.g. `srun bash -l -c ". $HOME/picongpu.profile"`)

Reply:
It is also on a Taurus node for a very small example
```
echo $BASH_SOURCE
```
When I `source` this file, it shows the correct path. I am still on it...

Reply:
as we used it for now, `BASH_SOURCE` is only available inside a sourced bash script.
this works on the head node at least, which is where we need it primarily.

Reply:
This is probably, why I never noticed that this is a real problem - it just worked. :laughing:

Reply:
Okay, I found the issue. Every `module` command for some reason deletes this bash variable on a KNL node - no idea why. Moving the definition of `$PIC_PROFILE` to the first line, solves this issue. :wink: